tests/foreman/api/test_subscription.py

```diff
@@ -177,15 +170,17 @@ def test_negative_upload():
         organization.
     """
     orgs = [entities.Organization().create() for _ in range(2)]
-    with manifests.clone() as manifest:
+    with function_entitlement_manifest as manifest:
```

Comment:
function_entitlement_manifest is already in contextmanager, I think Manifestor contextmanager would help here instead.
```suggestion
    with Manifester(manifest_category=settings.manifest.entitlement) as manifest:
```

Reply:
My thinking is that calling manifester directly in a test function should only be done in situations where it is required to do so (such as a single test function that requires multiple manifests). That way, we have a centralized location in the fixture definitions in case we need to make changes to how they operate in the future. For that reason, I would prefer to keep this as is. Could you please explain what you see as the functional advantage of making this change?

Reply:
@Gauravtalreja1 So this test actually requires that we use one manifest. It's testing to ensure that uploading the same manifest to two different organizations results in an error. I did run this test as written in this PR and it passes (i.e. we do get the error when the manifest is uploaded to the second org).

I agree that we could relatively easily search for the string `Manifester` to identify patterns like the one you're suggesting if we need to make updates in the future. However, the reason I'm pushing back on your suggestion is that, while I've been working on the manifester conversions, I have noticed some design patterns in Robottelo that perform ad hoc actions instead of using fixtures, similar to this case (the one that stands out the most is creating an organization in a test function with `entities.Organization.create()` instead of using the `function_org` fixture). While such patterns function perfectly well, in my opinion they add friction to the process of maintaining or refactoring code. This is most noticeable when making changes to a large number of tests like I'm doing with the manifester conversions. Modifying one instance of such a pattern is fairly trivial, but the work adds up when it has to be repeated dozens or hundreds of times.

To give an example of this type of change that's directly relevant here, at some point in the future, we will need to convert all uses of entitlement manifests to Simple Content Access manifests (because entitlement mode will eventually be removed as an option for manifests). To me, it seems simpler to primarily modify the fixtures being used by tests to implement that change and to reserve exceptions for cases in which calling Manifester directly is functionally required.

Let me know if you agree with this; if not, I'm still open to arguments for using the pattern that you suggested.

Reply:
@synkd Okay, got it, if this works then it's fine, won't block PR for this suggestion.